### ISSUE TITLE
Add better decoding support

### ISSUE THREAD

**Issue by Jegp** [norse/norse#34]
Similar to #27 but for decoding. We should have built-in defaults for common decoding schemes, like softmax, first spike, population, angular, etc.
Any others @cpehle ?

**Comment by cpehle** [comment]
This is connected to choices of loss functions. We should probably address to things: 

- Collect references to literature that contain spike decoding schemes and loss functions
- Implement those as pytorch modules
